# Aufgabe 24

## a)

Designmatrix aufstellen: 
\begin{equation}
A= \begin{pmatrix}
  \cos{(\frac{0}{6}\pi)} & \sin{(\frac{0}{6}\pi)} \\
  \cos{(\frac{1}{6}\pi)} & \sin{(\frac{1}{6}\pi)} \\
  \cos{(\frac{2}{6}\pi)} & \sin{(\frac{2}{6}\pi)} \\
  \cos{(\frac{3}{6}\pi)} & \sin{(\frac{3}{6}\pi)} \\
  \cos{(\frac{4}{6}\pi)} & \sin{(\frac{4}{6}\pi)} \\
  \cos{(\frac{5}{6}\pi)} & \sin{(\frac{5}{6}\pi)} \\
  \cos{(\frac{6}{6}\pi)} & \sin{(\frac{6}{6}\pi)} \\
  \cos{(\frac{7}{6}\pi)} & \sin{(\frac{7}{6}\pi)} \\
  \cos{(\frac{8}{6}\pi)} & \sin{(\frac{8}{6}\pi)} \\
  \cos{(\frac{9}{6}\pi)} & \sin{(\frac{9}{6}\pi)} \\
  \cos{(\frac{10}{6}\pi)} & \sin{(\frac{10}{6}\pi)} \\
  \cos{(\frac{11}{6}\pi)} & \sin{(\frac{11}{6}\pi)} \\
\end{pmatrix}
\end{equation}


## b)

In [2]:
import numpy as np


\begin{equation}
    \hat{a}=\left(A^{T}A \right)^{-1} A^{T} y
\end{equation}

In [3]:
A=np.matrix([[np.cos(0), np.sin(0)],
            [np.cos(1/6*np.pi), np.sin(1/6*np.pi)], 
            [np.cos(2/6*np.pi), np.sin(2/6*np.pi)], 
            [np.cos(3/6*np.pi), np.sin(3/6*np.pi)], 
            [np.cos(4/6*np.pi), np.sin(5/6*np.pi)],
            [np.cos(5/6*np.pi), np.sin(5/6*np.pi)], 
            [np.cos(6/6*np.pi), np.sin(6/6*np.pi)], 
            [np.cos(7/6*np.pi), np.sin(7/6*np.pi)], 
            [np.cos(8/6*np.pi), np.sin(8/6*np.pi)], 
            [np.cos(9/6*np.pi), np.sin(9/6*np.pi)], 
            [np.cos(10/6*np.pi), np.sin(10/6*np.pi)], 
            [np.cos(11/6*np.pi), np.sin(11/6*np.pi)]])

y=np.matrix([-0.032, 0.010, 0.057, 0.068, 0.076, 0.080, 0.031, 0.005, -0.041, -0.041 -0.090, -0.088, -0.074])

a=(A.T*A)**(-1)*A.T*y.T
a

matrix([[-0.04019568],
        [ 0.08817037]])

\begin{align}
    a_0=-0.0401956 \\
    a_1= 0.08817037
\end{align}

## C)

\begin{equation}
V[\hat{a}]= \sigma^2 \left(A^T A\right)
\end{equation}

In [4]:
sigma = 0.011
V = sigma**2 * (A.T * A).I
V

matrix([[ 2.01871558e-05, -6.71728350e-07],
        [-6.71728350e-07,  2.20223518e-05]])

In [5]:
np.sqrt(V[0,0]), np.sqrt(V[1,1])

(0.004493011885516234, 0.004692797863270538)

Also ergibt sich:
\begin{align}
    \sigma_{\alpha_1} &= 4,49 \cdot 10^{-3} \\
    \sigma_{\alpha_2} &= 4,69 \cdot 10^{-3}
\end{align}

In [6]:
rho=V[0, 1]/(np.sqrt(V[0,0]*V[1, 1]))
rho

-0.0318584221555165

womit
\begin{equation}
    \rho= \frac{\sigma_{1, 2}}{\sigma_1 \sigma_2} = \frac{-6,717 \cdot 10^{-7}}{\sqrt{2,019 \cdot 10^{-5} \cdot 2,202 \cdot 10^{-5}}}= -0,032
\end{equation}
folgt.


## d)

Es gilt:
\begin{equation}
    \cos{(\Psi+\delta)}=\cos{(\Psi)} \cos{(\delta)} - \sin{(\Psi)} \sin{(\delta)}
\end{equation}

Daraus folgt:
\begin{align*}
    a_1 &= A_0 \cdot \cos{\delta} \\
    a_2 &= - A_0 \cdot \sin{\delta} 
\end{align*}
und daraus wiederum
\begin{align*}
      {a_1}^2+{a_2}^2={A_0}^2 & \rightarrow A_0=\sqrt{{a_1}^2+{a_2}^2} \\
      \delta = - \arctan{\frac{a_2}{a_1}}
\end{align*}

Für die Fehler folgt dann:
\begin{align*}
      \sigma_{A_0}=\sqrt{\frac{{a_0}^2}{{a_1}^2+{a_0}^2} \sigma_{a_0}^2 + \frac{{a_1}^2}{{a_1}^2+{a_0}^2} \sigma_{a_1}^2 +2\frac{a_0 a_1}{\left({a_1}^2+{a_0}^2\right)} \cdot \text{Cov}(a_0, a_1) } \approx 0,0045\\
      \sigma_{\delta}=\sqrt{\frac{{a_0}^2}{\left({a_1}^2+{a_0}^2\right)^2} \sigma_{a_0}^2 + \frac{{a_1}^2}{\left({a_1}^2+{a_0}^2\right)^2} \sigma_{a_1}^2 -2\frac{a_0 a_1}{\left({a_1}^2+{a_0}^2\right)^2} \cdot \text{Cov}(a_0, a_1) } \approx 0,05
\end{align*}

\begin{align}
    A_0=0,0958 \pm 0,0045 \\ 
    \delta= 1.1450 \pm 0,0467
\end{align}